sycl/source/detail/config.hpp

```diff
@@ -176,6 +176,85 @@ template <> class SYCLConfig<SYCL_PI_TRACE> {
   }
 };
 
+template <> class SYCLConfig<SYCL_PARALLEL_FOR_RANGE_ROUNDING_TRACE> {
+  using BaseT = SYCLConfigBase<SYCL_PARALLEL_FOR_RANGE_ROUNDING_TRACE>;
+
+public:
+  static bool get() {
+    static bool Initialized = false;
+    static bool Trace = false; // No tracing by default
+
+    // Check env var setting for tracing only once
+    if (Initialized)
+      return Trace;
+
+    const char *ValStr = BaseT::getRawValue();
+    Trace = ValStr != nullptr;
+    Initialized = true;
+    return Trace;
+  }
+};
+
+template <> class SYCLConfig<SYCL_DISABLE_PARALLEL_FOR_RANGE_ROUNDING> {
+  using BaseT = SYCLConfigBase<SYCL_DISABLE_PARALLEL_FOR_RANGE_ROUNDING>;
+
+public:
+  static bool get() {
+    static bool Initialized = false;
+    static bool DisableRounding = false; // Range rounding on by default
+
+    // Check env var setting for rounding only once
+    if (Initialized)
+      return DisableRounding;
+
+    const char *ValStr = BaseT::getRawValue();
+    DisableRounding = ValStr != nullptr;
+    Initialized = true;
+    return DisableRounding;
+  }
+};
+
+template <> class SYCLConfig<SYCL_PARALLEL_FOR_RANGE_ROUNDING_PARAMS> {
+  using BaseT = SYCLConfigBase<SYCL_PARALLEL_FOR_RANGE_ROUNDING_PARAMS>;
+
+private:
+public:
+  static void GetSettings(size_t &MinFactor, size_t &GoodFactor,
+                          size_t &MinRange) {
+    static bool Initialized = false;
+    static bool NewFactors = false;
+    static size_t MF;
+    static size_t GF;
+    static size_t MR;
+    if (!Initialized) {
+      // Parse optional parameters of this form:
+      // MinRound:PreferredRound:MinRange
+      char *RoundParams = getenv("SYCL_PARALLEL_FOR_RANGE_ROUNDING_PARAMS");
+      if (RoundParams != nullptr) {
+        std::string Params(RoundParams);
+        size_t Pos = Params.find(':');
+        if (Pos != std::string::npos) {
+          MF = std::stoi(Params.substr(0, Pos));
+          Params.erase(0, Pos + 1);
+          Pos = Params.find(':');
+          if (Pos != std::string::npos) {
+            GF = std::stoi(Params.substr(0, Pos));
+            Params.erase(0, Pos + 1);
+            MR = std::stoi(Params);
+          }
+        }
+        NewFactors = true;
```

Comment:
It looks like NewFactors variable is redundand. If it is removed the behaviour is the same.

Reply:
No, NewFactors is not redundant. Only if new parameters are specified through the env var are the output variables MinFactor, GoodFactor and MinRange updated. If the env var is not set, the default values of those three variables remain unchanged in the calling function.

Reply:
In the other words the code update outputs parameters only if the environment variable is set.
This can be done using the following pseudo-code:
```
static char* EnvVal = getenv(...);
if(EnvVal != nullptr) {
   /// pars value from environment
   /// update output parameters
}
```

Reply:
The getenv calls are limited to one by using static char* RoundParams = ...
A flag is still needed to ensure that the getenv string is parsed only once.